src/lambda/wizards/samDeployWizard.ts

```diff
@@ -635,13 +685,26 @@ export class SamDeployWizard extends MultiStepWizard<SamDeployWizardResponse> {
 
     private readonly S3_BUCKET: WizardStep = async step => {
         const response = await this.context.promptUserForS3Bucket(step, this.response.region, this.response.s3Bucket)
-
-        this.response.s3Bucket = response
-
+        
         if (!response) {
             return WIZARD_GOBACK
         }
 
+        if(response === CREATE_NEW_BUCKET) {
+            const newBucketRequest = await this.context.promptUserForNewS3Bucket(step, this.response.region)
+            if (!newBucketRequest) {
+                return WIZARD_RETRY
+            }    
+            const newBucketName = await createNewS3Bucket(newBucketRequest, this.response.region!)
```

Comment:
Nit: why is one of the `this.response.region`s asserted and one not?

Reply:
One is an optional parameter and the other is required. While region can technically be undefined, this step will not be reached until it is defined.